File: src/main/java/macro/buddy/builds/UpdateGem.java
package macro.buddy.builds;

/**
 * Created by Macro303 on 2020-Jan-08.
 */
public class UpdateGem {
	private String oldGem;
	private String newGem;
	private String reason;

	public UpdateGem() {
	}

	public UpdateGem(String oldGem, String newGem, String reason) {
		this.oldGem = oldGem;
		this.newGem = newGem;
		this.reason = reason;
	}

	public String getOldGem() {
		return oldGem;
	}

	public void setOldGem(String oldGem) {
		this.oldGem = oldGem;
	}

	public String getNewGem() {
		return newGem;
	}

	public void setNewGem(String newGem) {
		this.newGem = newGem;
	}

	public String getReason() {
		return reason;
	}

	public void setReason(String reason) {
		this.reason = reason;
	}
}

File: src/main/kotlin/github/macro/build_info/gems/UpdateGem.kt
package github.macro.build_info.gems

import javafx.beans.property.SimpleStringProperty
import tornadofx.*

/**
 * Created by Macro303 on 2020-Jan-13.
 */
class UpdateGem() : JsonModelAuto {
	val oldGemProperty = SimpleStringProperty()
	var oldGem by oldGemProperty

	val newGemProperty = SimpleStringProperty()
	var newGem by newGemProperty

	val reasonProperty = SimpleStringProperty()
	var reason by reasonProperty

	constructor(oldGem: String, newGem: String, reason: String) : this() {
		this.oldGem = oldGem
		this.newGem = newGem
		this.reason = reason
	}

	override fun toString(): String {
		return "UpdateGem(oldGemProperty=$oldGemProperty, newGemProperty=$newGemProperty, reasonProperty=$reasonProperty)"
	}
}